src/checkout-buttons/strategies/apple-pay/apple-pay-button-strategy.ts

```diff
@@ -216,12 +217,25 @@ export default class ApplePayButtonStrategy implements CheckoutButtonStrategy {
         const { currency: { decimalPlaces } } = state.cart.getCartOrThrow();
         let checkout = state.checkout.getCheckoutOrThrow();
         const availableOptions = checkout.consignments[0].availableShippingOptions;
-        const shippingOptions = availableOptions?.map(option => ({
+        const selectedOption = availableOptions?.filter(option => option.id === checkout.consignments[0].selectedShippingOption?.id);
```

Comment:
Why not use `find` here to find the `selectedOption`, the you don't need to map over it to create the `firstShippingOption`?

Reply:
Hi @animesh1987, thanks a lot,  you are deadly right!!!, I will use your suggestion for sure.

Reply:
Hi @animesh1987, I do the changes to the code, please note that I still use the unselectedOptions, because I reorder the shippingOptions to make the first option the selected one.